### PR TITLE
[RFR] Updated “Facade Pattern” entry

### PR DESCRIPTION
**Pull request by virtyaluk**

### CODE REVIEW

File: glossary/FACADE_PATTERN.md [hunk 1]
```diff
@@ -1,3 +1,85 @@
...
+&nbsp;&nbsp;&nbsp;The objects are using the Facade Pattern to access resources from the Packages.
+
+## Example
+This is an abstract example of how a client ("you") interacts with a facade (the "computer") to a complex system (internal computer parts, like CPU and HardDrive).
+
+```javascript
+/* Complex parts */
+
+class CPU {
+	freeze() { /* code here */ }
```
Comment (KittyGiraudel) [inline glossary/FACADE_PATTERN.md]: Indentation is broken here.

File: glossary/FACADE_PATTERN.md [hunk 2]
```diff
@@ -1,3 +1,85 @@
...
+
+class HardDrive {
+    read(lba, size) { /* code here */ }
+}
+
+/* Facade */
+
+class ComputerFacade {
+	constructor() {
+		this.processor = new Cput();
```
Comment (KittyGiraudel) [inline glossary/FACADE_PATTERN.md]: Indentation is broken here.

File: glossary/FACADE_PATTERN.md [hunk 3]
```diff
@@ -1,3 +1,85 @@
...
+		this.ram = new Memory();
```
Comment (KittyGiraudel) [inline glossary/FACADE_PATTERN.md]: Indentation is broken here.

File: glossary/FACADE_PATTERN.md [hunk 4]
```diff
@@ -1,3 +1,85 @@
...
+        this.processor.execute();
+    }
+}
+
+
+/* Client */
+
+let computer = new ComputerFacade();
+computer.start();
+
```
Comment (KittyGiraudel) [inline glossary/FACADE_PATTERN.md:82]: One too many line here.

File: glossary/FACADE_PATTERN.md [hunk 5]
```diff
@@ -1,3 +1,85 @@
...
+The facade pattern is typically used when:
+
+- a simple interface is required to access a complex system;
+- the abstractions and implementations of a subsystem are tightly coupled;
+- need an entry point to each level of layered software; or
+- a system is very complex or difficult to understand.
+
+## Structure
+
+![](https://upload.wikimedia.org/wikipedia/en/5/57/Example_of_Facade_design_pattern_in_UML.png)
```
Comment (KittyGiraudel) [inline glossary/FACADE_PATTERN.md]: Please specify an alt text. :)

File: glossary/FACADE_PATTERN.md [hunk 6]
```diff
@@ -1,3 +1,85 @@
...
+
+**Facade**
+&nbsp;&nbsp;&nbsp;The facade class abstracts Packages 1, 2, and 3 from the rest of the application.
+**Clients**
+&nbsp;&nbsp;&nbsp;The objects are using the Facade Pattern to access resources from the Packages.
+
+## Example
+This is an abstract example of how a client ("you") interacts with a facade (the "computer") to a complex system (internal computer parts, like CPU and HardDrive).
```
Comment (KittyGiraudel) [inline glossary/FACADE_PATTERN.md:42]: Missing empty line here.

**Comment by virtyaluk** [comment]
Hey @HugoGiraudel, If you mean `&nbsp;` at line [36](https://github.com/HugoGiraudel/SJSJ/pull/117/files#diff-f316dfa402b83abaff45e7c03367bf97R36) and [38](https://g
ithub.com/HugoGiraudel/SJSJ/pull/117/files#diff-f316dfa402b83abaff45e7c03367bf97R38), I haven't found a way to make text indentation using pure markdown. But it's needed to make the visual perception of the article more pleasing.

I updated the file. If it's passed moderation then I want to add another article later today.

**Comment by KittyGiraudel** [comment]
Haha, I think we didn't understand each other. I’ll update the file, don’t worry. Thank you for your work as always! :)

On topic: if you would like to review (and complete?) existing entries instead of writing new ones, that would also be very appreciated. :)

**Comment by virtyaluk** [comment]
@HugoGiraudel

> I’ll update the file, don’t worry.

Thanks :)

> if you would like to review (and complete?) existing entries instead of writing new ones

First, I'd like to fill all the "_to be completed_" definitions and related glossary items. It's only 14 items (with this one). Then I've wanted to add few (maybe 10 :)) new terms, but yes, I can review existing terms before adding new ones. 

Once you merge this pull request I will continue filling out other terms.

**Pull request merged**